runtime/near-vm-runner/src/imports.rs

```diff
@@ -41,11 +57,18 @@ macro_rules! wrapped_imports {
             pub mod wasmer_ext {
                 use near_vm_logic::VMLogic;
                 use wasmer_runtime::Ctx;
+                use crate::imports::str_eq;
                 type VMResult<T> = ::std::result::Result<T, near_vm_logic::VMLogicError>;
                 $(
                     #[allow(unused_parens)]
                     $(#[cfg(feature = $feature_name)])*
                     pub fn $func( ctx: &mut Ctx, $( $arg_name: $arg_type ),* ) -> VMResult<($( $returns ),*)> {
+                        const IS_GAS: bool = str_eq(stringify!($func), "gas");
+                        if IS_GAS {
+                            None
+                        } else {
+                            Some(tracing::debug_span!(target: "host-function", stringify!($func)).entered())
+                        };
```

Comment:
Wait, this doesn't do what we want it to. This `;` immediately drops the span, *before* the host function is even invoked. We do need to bind it to a temporary variable to make sure it is dropped after `$func` returns: 

```rust
                        let _span = if IS_GAS {
                            None
                        } else {
                            Some(tracing::debug_span!(target: "host-function", stringify!($func)).entered())
                        };
```

Reply:
ah yes. Sorry

Reply:
Test again on testnet, with correctly set _span the time of certain host functions looks realistic!